src/helpers/internal_viewport.ts

```diff
@@ -203,18 +203,34 @@ export class InternalViewport {
     this.adjustViewportZoneY();
   }
 
+  /**
+   *
+   * @param zone
+   * @returns Computes the absolute coordinate of a given zone inside the viewport
+   */
   getRect(zone: Zone): Rect | undefined {
     const targetZone = intersection(zone, this.zone);
     if (targetZone) {
+      const x =
+        this.getters.getColRowOffset("COL", this.zone.left, targetZone.left) +
+        this.offsetCorrectionX;
+
+      const y =
+        this.getters.getColRowOffset("ROW", this.zone.top, targetZone.top) + this.offsetCorrectionY;
+
+      const width = Math.min(
+        this.getters.getColRowOffset("COL", targetZone.left, targetZone.right + 1),
+        this.width
```

Comment:
Does it work with freezed panes ?
`width` wouldn't take into account freezed cols and it would be smaller than the col offset ?
I think I'm missing something

Reply:
Actually, `getColRowOffset` should be named `getHeadersDistance` or `getDistanceBetweenHeaders`...

Reply:
This is a pane aka a viewport. A frozen pane will only imply the presence of more than one viewport . Each of them computes the size of a selected zone inside their own scope (hence the clipping with its own width) and each size is then assembled together in the getter getVisibleRect which is inside the plugin SheetView.